kuma/users/views.py

```diff
@@ -687,27 +670,21 @@ def form_valid(self, form):
         So, we need to manually commit the user to the db for it.
         """
         selected_email = form.cleaned_data["email"]
-        if form.other_email_used:
-            data = {
-                "email": selected_email,
-                "verified": False,
-                "primary": True,
-            }
-        else:
-            data = self.email_addresses.get(selected_email, None)
+        if selected_email not in self.email_addresses:
```

Comment:
As written, this doesn't work for signup via Google, because `self.email_addresses` is always an empty list for Google signups. In the `get_form` method above, the results for these two lines:
```python
        email = self.sociallogin.account.extra_data.get("email") or None
        email_data = (self.sociallogin.account.extra_data.get("email_addresses")) or []
```
are very different depending upon whether you signup with GitHub or Google. With GitHub, `email` will typically(always?) be `None` and `email_data` will be a non-empty list, while with Google it's the opposite, i.e. `email` will be the user's verified Google email and `email_data` will be an empty list.

I'm doubting that it still makes sense to keep the `self.default_email` idea, the distinction between `email` and `extra_email_addresses`. I'm wondering if we could just merge the `email` and `email_data` values into a single list of email addresses, use that for `self.email_addresses`, and drop `self.default_email`?

Also, right now we're just testing this with GitHub login, and it seems we should add an additional test for Google login since they're so different in their social-login data.

Reply:
I'll work on that. 

I'm disgusted that my change would have broken logging in with Gmail and I didn't notice. I.e. that the tests passed. Not sure if we're mocking incorrectly or we're not even testing it. 

I know I tested signing up with Gmail in this branch in my local browser. But perhaps I did  some code refactorings *after* having done a full end-to-end testing. 
Gross!

Reply:
Oh it gets worse!
I know I kinda rushed this PR on Friday when all I had was a half-day but guess what; not only did I managed to break signup for Google, it also breaks for GitHub!!

So, that means that my (bad) code changes, which had great intentions, broke both Google AND GitHub and none of our tests noticed.